app/src/androidTest/kotlin/at/bitfire/davdroid/repository/TestDavCollectionRepository.kt

```diff
@@ -0,0 +1,80 @@
+package at.bitfire.davdroid.repository
+
+import android.content.Context
+import at.bitfire.davdroid.db.AppDatabase
+import at.bitfire.davdroid.db.Collection
+import at.bitfire.davdroid.db.Service
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import io.mockk.junit4.MockKRule
+import kotlinx.coroutines.test.runTest
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import javax.inject.Inject
+
+@HiltAndroidTest
+class TestDavCollectionRepository {
+
+    @get:Rule
+    var hiltRule = HiltAndroidRule(this)
+    @get:Rule
+    val mockkRule = MockKRule(this)
+
+    @Inject
+    lateinit var db: AppDatabase
+
+    @Inject
+    lateinit var collectionRepository: DavCollectionRepository
+
+    var service: Service? = null
+
+    @Before
+    fun setUp() {
+        hiltRule.inject()
+        service = createTestService(Service.TYPE_CARDDAV)!!
+    }
+
+    @After
+    fun cleanUp() {
+        db.close()
+    }
+
+    @Test
+    fun testAddOnChangeListener() = runTest {
```

Comment:
This test tests whether setForceReadOnly calls the callback (plus `addOnChange` because otherwise it can't be called, but that doesn't seem to be the "main goal" of the test). So I'd rename it accordingly.